### PR TITLE
fixes controller e2e flapping

### PR DESCRIPTION
**Pull request by xh3b4sd**
Fixes https://circleci.com/gh/giantswarm/operatorkit/2617.

### CODE REVIEW

File: controller/integration/test/reconciliation/reconciliation_test.go [hunk 1]
```diff
@@ -101,20 +101,23 @@ func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
...
-	// Setting the labels on createdObj works through the magic or accessors and
-	// pointers here.
-	_, err = testWrapper.UpdateObject(testNamespace, createdObj)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
+	{
+		o := func() error {
+			createdNodeConfig.SetLabels(map[string]string{"testlabel": "testlabel"})
+
+			_, err = nodeConfigWrapper.UpdateObject(testNamespace, createdNodeConfig)
```
Comment (xh3b4sd) [inline controller/integration/test/reconciliation/reconciliation_test.go]: Part of the fix is to use a backoff here.

File: controller/integration/wrapper/nodeconfig/crud.go [hunk 2]
```diff
@@ -42,11 +42,18 @@ func (w Wrapper) GetObject(name, namespace string) (interface{}, error) {
...
 	if err != nil {
-		return nil, err
+		return nil, microerror.Mask(err)
 	}
+
+	m, err := w.g8sClient.CoreV1alpha1().NodeConfigs(namespace).Get(nodeConfig.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	nodeConfig.SetResourceVersion(m.GetResourceVersion())
```
Comment (xh3b4sd) [inline controller/integration/wrapper/nodeconfig/crud.go]: Another part of the fix is to fetch the latest resource version.

**Comment by MarcelMue** [comment]
I took the liberty to remove the required thingy from all the non running tests. Feel free to add them back later.

**Comment by xh3b4sd** [comment]
It looks like we have more flapping tests out of a sudden. I merge this here because one test should be better already. I will check and get back on the others.

**Pull request merged**